### PR TITLE
Feature/tagging resources

### PR DESCRIPTION
**Pull request by SchneiderNicolas**
This PR adds the route in API to get a list of resources. This route sends an ES request and return the list of resources in a JSON format.
You can also filter resources and get them by region and resources type.

### CODE REVIEW

File: resourcesTagging/prepareResponse.go [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+package resourcesTagging
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/olivere/elastic"
+	"github.com/trackit/jsonlog"
+	"github.com/trackit/trackit/tagging/utils"
```
Comment (alexandre-snr) [inline resourcesTagging/prepareResponse.go]: New line misplaced

File: resourcesTagging/routes.go [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+
+package resourcesTagging
+
+import (
+	"net/http"
+	"database/sql"
+	"github.com/trackit/trackit/db"
```
Comment (alexandre-snr) [inline resourcesTagging/routes.go]: Missing new line

**Review by alexandre-snr — COMMENTED** [review]
Not much to say, waiting for Simon and Jules' approvals

File: resourcesTagging/getResourcesData.go [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+		return returnCode, nil, err
+	} else if res == nil {
+		return http.StatusInternalServerError, nil, errors.New("Error while getting data. Please check again in few hours.")
+	}
+	resources, err := prepareResponseResources(ctx, res)
+	if err != nil {
+		return  http.StatusInternalServerError, nil, err
+	}
+	return http.StatusOK, resources, nil
+}
```
Comment (simonmeyerrr) [inline resourcesTagging/getResourcesData.go]: missing new line

File: resourcesTagging/prepareResponse.go [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+	for _, account := range response.Accounts.Buckets {
+		for _, date := range account.Dates.Buckets {
+			for _, resource := range date.Resources.Hits.Hits {
+				resources = append(resources, resource.Resource)
+			}
+		}
+
+	}
+	return resources, nil
+}
```
Comment (simonmeyerrr) [inline resourcesTagging/prepareResponse.go]: missing new line

File: resourcesTagging/routes.go [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+	if a[resourcesQueryArgs[2]] != nil {
+		parsedParams.ResourceType = a[resourcesQueryArgs[2]].([]string)
+	}
+	returnCode, report, err := GetResourcesData(request.Context(), parsedParams, user, tx)
+	if err != nil {
+		return returnCode, err
+	} else {
+		return returnCode, report
+	}
+}
```
Comment (simonmeyerrr) [inline resourcesTagging/routes.go]: missing new line

File: server/server.go [hunk 6]
```diff
@@ -36,6 +36,7 @@ import (
 	"github.com/trackit/trackit/periodic"
 	_ "github.com/trackit/trackit/plugins"
 	_ "github.com/trackit/trackit/reports"
+	_ "github.com/trackit/trackit/resourcesTagging"
```
Comment (simonmeyerrr) [inline server/server.go]: you should put your code in existing package `tagging/routes`

File: resourcesTagging/routes.go [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+	"github.com/trackit/trackit/routes"
+	"github.com/trackit/trackit/users"
+)
+
+type (
+	// ResourcesQueryParams will store the parsed query params
+	ResourcesQueryParams struct {
+		AccountList  []string
+		IndexList    []string
+		Region       []string
```
Comment (simonmeyerrr) [inline resourcesTagging/routes.go]: Add a `s` for all variables that contain arrays to better identify them

**Review by simonmeyerrr — CHANGES_REQUESTED** [review]

File: resourcesTagging/esRequestConstructor.go [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+//createQueryTypeFilterResources creates and return a nex *elastic.TermsQuery on the typeList array
+func createQueryTypeFilterResources(typeList []string) *elastic.TermsQuery {
+	typeListFormatted := make([]interface{}, len(typeList))
+	for i, v := range typeList {
+		typeListFormatted[i] = v
+	}
+	return elastic.NewTermsQuery("resourceType", typeListFormatted...)
+}
+
+
```
Comment (jklakosz) [inline resourcesTagging/esRequestConstructor.go]: 2 new lines, only 1 is good enough

**Review by jklakosz — CHANGES_REQUESTED** [review]

File: tagging/routes/prepareResponse.go [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+package routes
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/olivere/elastic"
+	"github.com/trackit/jsonlog"
+	"github.com/trackit/trackit/tagging/utils"
+
```
Comment (jklakosz) [inline tagging/routes/prepareResponse.go:23]: this line space must be at 23rd line

**Review by alexandre-snr — APPROVED** [review]

**Review by jklakosz — APPROVED** [review]

File: tagging/routes/routes.go [hunk 10]
```diff
@@ -28,6 +28,23 @@ var taggingComplianceQueryArgs = []routes.QueryArg{
 	routes.DateEndQueryArg,
 }
 
+// resourcesQueryArgs allows to get required queryArgs params
+var resourcesQueryArgs = []routes.QueryArg{
+	routes.AwsAccountsOptionalQueryArg,
+	routes.QueryArg{
+		Name:        "region",
```
Comment (simonmeyerrr) [inline tagging/routes/routes.go]: missing `s`

File: tagging/routes/routes.go [hunk 11]
```diff
@@ -28,6 +28,23 @@ var taggingComplianceQueryArgs = []routes.QueryArg{
...
+		Type:        routes.QueryArgStringSlice{},
+		Description: "The region of the resource",
+		Optional:    true,
+	},
+	routes.QueryArg{
+		Name:        "resourceType",
```
Comment (simonmeyerrr) [inline tagging/routes/routes.go]: missing `s`

**Review by simonmeyerrr — CHANGES_REQUESTED** [review]

**Review by simonmeyerrr — APPROVED** [review]

**Pull request merged**